mig-agent/config.go

```diff
@@ -52,54 +52,205 @@ func configLoad(path string) (err error) {
 	if err = gcfg.ReadFileInto(&config, path); err != nil {
 		panic(err)
 	}
-	hbf, err := time.ParseDuration(config.Agent.HeartbeatFreq)
-	if err != nil {
+	var globals = NewGlobals()
+	if err = globals.parseConfig(config); err != nil {
 		panic(err)
 	}
-	timeout, err := time.ParseDuration(config.Agent.ModuleTimeout)
-	if err != nil {
-		panic(err)
+	return
+}
+
+// Globals receives parsed config settings and applies them to global vars.
+// NewGlobals returns a Globals struct populated with initial values from global vars.
+type Globals struct {
```

Comment:
Does this type and all these members need to be exported? Or should this be "globals", etc ?